yourbench/pipeline/answer_generation.py

```
@@ -0,0 +1,276 @@
+# yourbench/pipeline/answer_generation.py
+
+"""
+answer_generation.py
+
+Stage: answer_generation
+------------------------
+Generates answers to both single-shot and multi-hop questions for one or more 'scenarios'.
+Each scenario is appended to the *same* output subsets:
+  - single_shot_questions_with_answers
+  - multi_hop_questions_with_answers
+
+We distinguish the scenario by the 'answer_fashion' column in the final dataset (e.g. 'zero_shot', 'with_correct_chunk').
+
+Configuration Example (YAML):
+  answer_generation:
+    run: true
+    scenarios:
+      - zero_shot
+      - with_correct_chunk
+"""
+
+from typing import Any, Dict, List
+
+from loguru import logger
+
+from datasets import Dataset
+from yourbench.utils.prompts import GOLD_QA_USER_PROMPT, ZEROSHOT_QA_USER_PROMPT
+from yourbench.utils.dataset_engine import custom_load_dataset, custom_save_dataset
+from yourbench.utils.parsing_engine import extract_content_from_xml_tags
+from yourbench.utils.inference_engine import InferenceCall, run_inference
+
+
+def run(config: Dict[str, Any]) -> None:
+    """
+    Main entry point for the answer_generation stage.
+
+    Looks at config["pipeline"]["answer_generation"]["scenarios"] to see which answer-generation
+    modes to run. Currently supported: "zero_shot" and "with_correct_chunk".
+
+    Always concatenates new rows into:
+      - single_shot_questions_with_answers
+      - multi_hop_questions_with_answers
+
+    A column 'answer_fashion' indicates which scenario was used to generate the answer.
+    """
+    stage_cfg = config.get("pipeline", {}).get("answer_generation", {})
```

A possible situation here, if the config structure is wrong, it could silently fail or behave incorrectly. I would add a "TODO" here to integrate a config validation for the whole project

---

added